packages/trader/src/RangeTrader.ts

```diff
@@ -138,9 +138,12 @@ export class RangeTrader {
         ...commonLogObject,
         error: tradeExecutionTransaction
       });
-      return;
+      throw tradeExecutionTransaction;
     }
-    const exchangeSpotPriceAfterTrade = await this.exchangeAdapter.getExchangeSpotPrice();
+
+    // Get the post trade spot price to double check deviation error.
+    await this.tokenPriceFeed.update();
```

Comment:
refactored to use the price feed over the exchange adapter method.

Reply:
+1